packages/scandipwa/src/route/MenuPage/MenuPage.container.js

```diff
@@ -53,14 +58,18 @@ export class MenuPageContainer extends PureComponent {
     }
 
     componentDidUpdate() {
-        this.redirectIfNotOnMobile();
+        const { isMounted } = this.state;
+
+        if (isMounted) {
+            this.redirectIfNotOnMobile();
+        }
     }
```

Comment:
Why need to make it on didUpdate and check if it is mounted? Can't u actually make redirect on constructor?